CMakeLists.txt

```diff
@@ -66,49 +78,76 @@ else()
         -Wpedantic
     )
 endif()
-
-
-string (REPLACE ";" " " DIAGNOSTIC_FLAGS "${DIAGNOSTIC_FLAGS}")
-
-SET(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} ${DIAGNOSTIC_FLAGS}")
-
-find_package(GTest CONFIG REQUIRED)
-
-macro(add_test_dependencies TEST_TARGET)
-    target_link_libraries(${TEST_TARGET} GTest::gtest GTest::gtest_main)
-    target_include_directories(${TEST_TARGET} PRIVATE include)
-    add_test(NAME ${TEST_TARGET} COMMAND ${TEST_TARGET})
-endmacro()
-
-add_executable(comparison_chain_test test/comparison_chain_test.cpp)
-add_test_dependencies(comparison_chain_test)
-add_executable(concepts_test test/concepts_test.cpp)
-add_test_dependencies(concepts_test)
-add_executable(enum_array_test test/enum_array_test.cpp)
-add_test_dependencies(enum_array_test)
-add_executable(enum_map_test test/enum_map_test.cpp)
-add_test_dependencies(enum_map_test)
-add_executable(enum_set_test test/enum_set_test.cpp)
-add_test_dependencies(enum_set_test)
-add_executable(enum_utils_test test/enum_utils_test.cpp)
-add_test_dependencies(enum_utils_test)
-add_executable(fixed_map_test test/fixed_map_test.cpp)
-add_test_dependencies(fixed_map_test)
-add_executable(fixed_map_perf_test test/fixed_map_perf_test.cpp)
-add_test_dependencies(fixed_map_perf_test)
-add_executable(fixed_red_black_tree_test test/fixed_red_black_tree_test.cpp)
-add_test_dependencies(fixed_red_black_tree_test)
-add_executable(fixed_set_test test/fixed_set_test.cpp)
-add_test_dependencies(fixed_set_test)
-add_executable(fixed_vector_test test/fixed_vector_test.cpp)
-add_test_dependencies(fixed_vector_test)
-add_executable(index_range_predicate_iterator_test test/index_range_predicate_iterator_test.cpp)
-add_test_dependencies(index_range_predicate_iterator_test)
-add_executable(instance_counter_test test/instance_counter_test.cpp)
-add_test_dependencies(instance_counter_test)
-add_executable(pair_view_test test/pair_view_test.cpp)
-add_test_dependencies(pair_view_test)
-add_executable(string_literal_test test/string_literal_test.cpp)
-add_test_dependencies(string_literal_test)
-add_executable(type_name_test test/type_name_test.cpp)
-add_test_dependencies(type_name_test)
+add_library(project_warnings INTERFACE)
+target_compile_options(project_warnings INTERFACE ${DIAGNOSTIC_FLAGS})
+
+
+find_package(magic_enum CONFIG)
+find_package(range-v3 CONFIG)
```

Comment:
`range-v3` is only needed for tests, so it can be placed in the test-only region.